src/shims/tls.rs

```diff
@@ -203,11 +259,11 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             this.run()?;
         }
 
-        // Now run the "keyed" destructors.
-        let mut dtor = this.machine.tls.fetch_tls_dtor(None);
+        assert!(this.has_terminated(thread_id)?, "running TLS dtors for non-terminated thread");
+        let mut dtor = this.machine.tls.fetch_tls_dtor(None, thread_id);
         while let Some((instance, ptr, key)) = dtor {
-            trace!("Running TLS dtor {:?} on {:?}", instance, ptr);
-            assert!(!this.is_null(ptr).unwrap(), "data can't be NULL when dtor is called!");
+            trace!("Running TLS dtor {:?} on {:?} at {:?}", instance, ptr, thread_id);
+            assert!(!this.is_null(ptr).unwrap(), "Data can't be NULL when dtor is called!");
```

Comment:
```suggestion
            assert!(!this.is_null(ptr).unwrap(), "data can't be NULL when dtor is called!");
```
We usually follow rustc and start lower-case in error message and assertions.